components/ai_chat/core/browser/conversation_driver.h

```diff
@@ -110,31 +117,44 @@ class ConversationDriver {
  protected:
   virtual GURL GetPageURL() const = 0;
   virtual std::u16string GetPageTitle() const = 0;
-  virtual void GetPageContent(
-      base::OnceCallback<void(std::string, bool is_video)> callback) const = 0;
-  virtual bool HasPrimaryMainFrame() const = 0;
-  virtual bool IsDocumentOnLoadCompletedInPrimaryMainFrame() const = 0;
-
-  virtual void OnFaviconImageDataChanged();
 
-  void MaybeGeneratePageText();
-  void CleanUp();
+  // Implementer should fetch content from the "page" associated with this
+  // conversation.
+  // |is_video| lets the conversation know that the content is focused on video
+  // content so that various UI language can be adapted.
+  // |invalidation_token| is an optional parameter received in a prior callback
+  // response of this function against the same page. See GetPageContentCallback
+  // for explanation.
+  virtual void GetPageContent(GetPageContentCallback callback,
+                              std::string_view invalidation_token) const = 0;
```

Comment:
is `invalidation_token` a url essentially? how is it derived?

Reply:
derived? They are referring to the same thing. It's called `content_invalidation_token_` in ConversationDriver and `invalidation_token` in the`page_content_fetcher`.

Reply:
where exactly is it being generated?

Reply:
in this PR in the page content fetcher

Reply:
```
  // |invalidation_token| is an optional parameter that will be passed back on
  // the next call to |GetPageContent| so that the implementer may determine if
  // the page content is static or if it needs to be fetched again. Most page
  // content should be fetched again, but some pages are known to be static
  // during their lifetime and may have expensive content fetching, e.g. videos
  // with transcripts fetched over the network.
```